lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2039 @@
+      description:
+        - This is the main variable used for creating and updating distributions
+          and streaming distributions. When used, it will be a complex data
+          type as a dictionary that represents the config of the distribution.
+          When used for creating a distribution, it must contain at least one
+          origin in C(origins) or the parameter C(default_domain_name_origin)
+          as a minimum.
+          When used for creating a streaming distribution, it must contain
+          either C(s3_origin) or C(default_s3_origin_domain_name) as a
+          minimum.
+          Components of C(config) can be specified entirely in
+          C(config) or as separate elements outside of the config.
+          This parameter applies to both distributions and streaming
+          distributions.
+          Elements of a distribution are
+            C(caller_reference)
+            C(aliases[])
+            C(default_root_object)
+            C(origins[])
+            C(default_cache_behavior)
+            C(cache_behaviors[])
+            C(custom_error_responses[])
+            C(comment)
+            C(logging)
+            C(price_class)
+            C(enabled)
+            C(viewer_certificate)
+            C(restrictions)
+            C(web_acl_id)
+            C(http_version)
+            C(is_ipv6_enabled)
+          Elements of a streaming distribution are
+            C(caller_reference)
+            C(s3_origin)
+            C(aliases[])
+            C(comment)
+            C(logging)
+            C(trusted_signers)
+            C(http_version)
+            C(is_ipv6_enabled)
+          Most of these elements have sub-elements that can be seen in their
+          entirety in the boto3 documentation at
+            U(http://boto3.readthedocs.io/en/latest/reference/services/cloudfront.html#CloudFront.Client.create_distribution)
+          and
+            U(http://boto3.readthedocs.io/en/latest/reference/services/cloudfront.html#CloudFront.Client.create_streaming_distribution).
+          When element parameters are specified as well as the C(config)
+          parameter, the elements specified will have precendence and overwrite
+          any relevant data for that element in the C(config) variable.
+      required: false
+
+    tags:
+      description:
+        - Used for distributions and streaming distributions.
+          Should be input as a I(list[]) of I(Key) I(Value) pairs.
+      required: false
+
+    purge_tags:
+      description:
+        - Specifies whether existing tags will be removed before adding
+          new tags.
+          When C(purge_tags=yes), existing tags are removed and C(tags) are
+          added, if specified. If no tags are specified, it removes all
+          existing tags for the distribution.
+          When C(purge_tags=no), existing tags are kept and C(tags) are added,
+          if specified.
+      required: false
+      default: 'no'
+      choices: ['yes', 'no']
```

Comment:
I appreciate the feedback, but I'm not sure that's an entirely safe default to have - any modifications to a distribution are going to purge all tags regardless of whether tags have been specified or not. This would happen even for a small task to modify an existing distribution, where it would unwittingly remove all existing tags. 

We could rename it to `keep_tags` and have a default of yes if that would satisfy the default=yes requirement...